### PR TITLE
Feat/icons generate paths

### PR DESCRIPTION
**Pull request by literat**
Generate SVG paths as JS object to use them dynamically in React component.

The script will generate `constants/icons.js` with JS object containing the name of the icon as a key and value with SVG path content.

### CODE REVIEW

**Review by tomassychra — APPROVED** [review]

File: packages/icons/scripts/buildConstants.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    if (svgs.length > 0) {
+      const icons = {};
+      let distContent = 'const icons = ';
+
+      svgs.forEach((svg) => {
+        const iconName = svg.replace('.svg', '');
+        const svgPath = path.join(srcDir, svg);
+        const svgContent = fs.readFileSync(svgPath, 'utf8');
+        const dom = new jsdom.JSDOM(svgContent);
+        const pathContent = dom.window.document.querySelector('path').getAttribute('d');
```
Comment (crishpeen) [inline packages/icons/scripts/buildConstants.js:22]:
```suggestion
        icons[iconName] = dom.window.document.querySelector('svg').innerHTML;
```

We should store all children, not just one path

File: packages/icons/src/react/SvgIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+interface IconProps {
+  name: string;
+  size: number;
+}
+
+export const SvgIcon = ({ name, size, ...restProps }: IconProps): JSX.Element => {
+  return (
+    <svg viewBox="0 0 24 24" fill="currentColor" width={size || '24'} height={size || '24'} {...restProps}>
+      <path d={icons[name]} fill="currentColor" />
```
Comment (crishpeen) [inline packages/icons/src/react/SvgIcon.tsx:12]: and here should print all the children from the constants/SVG. CurrentColor is already in there.

File: packages/icons/package.json [hunk 3]
```diff
@@ -16,16 +16,19 @@
...
     "build:react:clean": "shx rm -rf dist/react/**.d.ts dist/react/**.js.map dist/.tmp-svg",
+    "build:const": "node ./scripts/buildConstants.js",
     "prebuild": "npm-run-all build:clean build:prepare",
-    "build": "npm-run-all build:svg prepare:react build:react",
-    "postbuild": "yarn build:react:clean"
+    "build": "npm-run-all build:svg prepare:react build:react build:const",
+    "postbuild": "npm-run-all build:react:clean build:copy"
   },
   "devDependencies": {
+    "jsdom": "^20.0.0",
```
Comment (crishpeen) [inline packages/icons/package.json:29]: jsdom to dependencies (not devD)?

**Review by crishpeen — CHANGES_REQUESTED** [review]

File: packages/icons/scripts/buildConstants.js (hunk 1 above)
Comment (literat) [reply packages/icons/scripts/buildConstants.js]: Ok, I've fixed this.

File: packages/icons/src/react/SvgIcon.tsx (hunk 2 above)
Comment (literat) [reply packages/icons/src/react/SvgIcon.tsx]: This is deprecated. I have chosen a better way to work with. See the follow-up PR.

File: packages/icons/package.json (hunk 3 above)
Comment (literat) [reply packages/icons/package.json:29]: JSDOM is used only for generating icons, so it is a devDependency.

File: packages/icons/README.md [hunk 4]
```diff
@@ -48,12 +48,33 @@ import Warning from '@lmc-eu/spirit-icons/svg/warning.svg';
...
+// or
+import WarningIcon from '@lmc-eu/spirit-icons/react/WarningIcon';
 
-<Warning />;
+<WarningIcon />;
+```
+
+### Icons paths
+
+Alternatively you can use an `icons` object which consists of an icon name and SVG path. Thus you can fabricate your icon yourself.
```
Comment (crishpeen) [inline packages/icons/README.md:63]:
```suggestion
Alternatively, you can use an `icons` object which consists of an icon name and SVG content. Thus you can fabricate your icon yourself.
```

File: packages/icons/scripts/buildConstants.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    if (svgs.length > 0) {
+      const icons = {};
+      let distContent = 'const icons = ';
+
+      svgs.forEach((svg) => {
+        const iconName = svg.replace('.svg', '');
+        const svgPath = path.join(srcDir, svg);
+        const svgContent = fs.readFileSync(svgPath, 'utf8');
+        const dom = new jsdom.JSDOM(svgContent);
+        const pathContent = dom.window.document.querySelector('svg').innerHTML.replaceAll('\n', '');
```
Comment (crishpeen) [inline packages/icons/scripts/buildConstants.js:22]: Maybe a different name? It's not only path content. It's whole SVG content.

**Review by crishpeen — APPROVED** [review]

File: packages/icons/scripts/buildConstants.js (hunk 5 above)
Comment (literat) [reply packages/icons/scripts/buildConstants.js]: Renamed.

File: packages/icons/README.md (hunk 4 above)
Comment (literat) [reply packages/icons/README.md]: Applied.

**Pull request merged**